core-service/src/main/java/javasabr/mqtt/service/publish/handler/impl/Qos1MqttPublishOutMessageHandler.java

```diff
@@ -21,11 +29,37 @@ public QoS qos() {
   }
 
   @Override
-  protected boolean handleReceivedResponse(NetworkMqttUser user, TrackableMqttMessage response) {
-    if (!(response instanceof PublishAckMqttInMessage)) {
-      throw new IllegalStateException("Unexpected response: " + response);
+  protected boolean handleReceivedTrackableMessageImpl(
+      ExternalNetworkMqttUser user, 
+      MqttSession session,
+      TrackableMqttMessage message,
+      @Nullable TrackedMessageMeta trackedMessageMeta) {
+    
+    int messageId = message.messageId();
+    String clientId = user.clientId();
+    if (trackedMessageMeta == null) {
+      log.warning(clientId, messageId, "[%s] No any stored information for messageId:[%d]"::formatted);
+      return true;
+    } else if (trackedMessageMeta.messageType() != MqttMessageType.PUBLISH) {
+      log.warning(clientId, trackedMessageMeta, messageId,
+          "[%s] No expected message meta:[%s] for messageId:[%d]"::formatted);
+      return true;
     }
-    // just return 'true' to remove pending packet from session
+    if (!(message instanceof PublishAckMqttInMessage publishAck)) {
```

Comment:
It might make sense to use `} else if {` branch here

Reply:
@crazyrokr then the reference to publishAck will not be available